source/material_model/rheology/visco_plastic.cc

```diff
@@ -256,12 +257,17 @@ namespace aspect
             // Step 3b: calculate current (viscous or viscous + elastic) stress magnitude
             double current_stress = 2. * viscosity_pre_yield * current_edot_ii;
 
-            // Step 4: calculate strain-weakened friction, cohesion
+            // Step 4a: calculate strain-weakened friction, cohesion
             const DruckerPragerParameters drucker_prager_parameters = drucker_prager_plasticity.compute_drucker_prager_parameters(j,
                                                                       phase_function_values,
                                                                       n_phases_per_composition);
             const double current_cohesion = drucker_prager_parameters.cohesion * weakening_factors[0];
-            const double current_friction = drucker_prager_parameters.angle_internal_friction * weakening_factors[1];
+            output_parameters.current_friction_angles[j] = drucker_prager_parameters.angle_internal_friction * weakening_factors[1];
+
+            // Steb 4b: calculate friction angle dependent on strain rate if specified
+            if (friction_options.get_friction_dependence_mechanism() == dynamic_friction)
+              output_parameters.current_friction_angles[j] = friction_options.compute_dependent_friction_angle(current_edot_ii,
```

Comment:
I find it confusing that the current_friction is no longer the current friction when a friction_dependence_mechanism is active. Maybe you can first compute the right current_friction and then store the final product in the output_parameters. Then you also don't have to use output_parameters below.

Reply:
yes sure, I can do this. I thought it would be good to remove unnecessary variables, but if you think that is more intuitive I am happy to change it back

Reply:
@anne-glerum  can you have a quick look at the new commit if this is better now?